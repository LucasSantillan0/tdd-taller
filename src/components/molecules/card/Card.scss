.card {
  width: fit-content;
  position: relative;
  &__img {
    max-width: 200px;
    max-height: 350px;
  }
  &__delete-button {
    width: fit-content;
    position: absolute;
    top: 10%;
    left: 120px;
    background-color: transparent;
    background-repeat: no-repeat;
    border: none;
    overflow: hidden;
    outline: none;
    cursor: pointer;
  }
  &__settings {
    height: 100%;
    width: 100%;
    position: absolute;
    top: 0%;
    color: white;
    display: flex;
    justify-content: center;
    align-items: center;
    flex-direction: column;
    background-color: rgba(0, 0, 0, 0.229);
    gap: 20px;
    z-index: 25;
    &__buttons {
      display: flex;
      flex-direction: column;
      gap: 20px;
    }
  }
}
