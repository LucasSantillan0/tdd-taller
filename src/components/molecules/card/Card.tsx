import { FC, useState } from "react";
import Button from "../../atmos/button/Button";
import Gif from "../../../utils/interfaces/gif";
import deleteIcon from "../../../assets/delete-icon.svg";
import "./Card.scss";
interface CardProps {
  handleDelete: (gif: Gif) => Promise<void>;
  gif: Gif;
}

const Card: FC<CardProps> = ({ handleDelete, gif }) => {
  const [isShowingOptions, setIsShowingOptions] = useState<boolean>(false);

  return (
    <article className="card">
      {isShowingOptions && (
        <div className="card__settings">
          <Button onClick={() => handleDelete(gif)}> Eliminar</Button>
          <Button onClick={() => setIsShowingOptions(false)}> Cancelar</Button>
        </div>
      )}
      <Button
        onClick={() => setIsShowingOptions(true)}
        className="card__delete-button"
        variant="secondary"
        type="button"
      >
        <img src={deleteIcon} alt={`showDelete-${gif.id}`} />
      </Button>
      <img src={gif.url} alt={`gif-${gif.id}`} className="card__img" />
    </article>
  );
};

export default Card;
