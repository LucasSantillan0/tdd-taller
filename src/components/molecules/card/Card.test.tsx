import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Card from "./Card";

describe("Card tests", () => {
  test("Should render the card", () => {});
  test("Should render buttons", async () => {});
  test("Should delete", async () => {});
});
