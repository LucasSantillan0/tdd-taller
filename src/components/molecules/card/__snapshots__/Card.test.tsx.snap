// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card tests Should render the card 1`] = `
<DocumentFragment>
  <div>
    <button
      class="button button--primary"
    >
      <img
        alt="showDelete-test"
        src=""
      />
    </button>
    <img
      alt="gif-test"
      src=""
    />
  </div>
</DocumentFragment>
`;
